libraries/resource_providers/api500/c7000/volume_template_provider.rb

```diff
@@ -0,0 +1,37 @@
+# (c) Copyright 2017 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+module OneviewCookbook
+  module API500
+    module C7000
+      # VolumeTemplate API500 C7000 provider
+      class VolumeTemplateProvider < API300::C7000::VolumeTemplateProvider
+        def load_resource_with_associated_resources
+          validate_presence_of(:storage_system, :storage_pool)
+          storage_system = load_storage_system
```

Comment:
How about:
```ruby
storage_system_data = {hostname: @new_resource.storage_system, name: @new_resource.storage_system}
storage_system  = load_resource(:StorageSystem, storage_system_data )
```

And drop the `load_storage_system` method? I'm not seeing the need for it.

Also, if we just want the storage system's URI and nothing else, we can use:
```ruby
storage_system_uri  = load_resource(:StorageSystem, storage_system_data, :uri)
```

Reply:
>`storage_system_uri  = load_resource(:StorageSystem, storage_system_data, :uri)` 

Nice!

Reply:
but we need get entire storage_system, to call one of its instance methods.